analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpMetrics.cs

```diff
@@ -79,12 +79,12 @@ protected override bool IsFunction(SyntaxNode node)
                 case SyntaxKind.ConstructorDeclaration:
                 case SyntaxKind.ConversionOperatorDeclaration:
                 case SyntaxKind.DestructorDeclaration:
-                case SyntaxKind.MethodDeclaration:
                 case SyntaxKind.OperatorDeclaration:
+                    return ((BaseMethodDeclarationSyntax)node).HasBodyOrExpressionBody(); // Non-abstract, non-interface methods
```

Comment:
I'm not sure I understand why we need to check for methods that the location is mapped correctly but not for the operators

Reply:
Can't we remove this call and use the implementation from the method declaration?

Reply:
I thought that `MethodDeclaration`s are the only one that might be generated by razor (see `BuildRenderTree` method). Everything else should have a mapped location in the original file and thus should be counted. I was thinking about doing the razor check only for that to minimize the impact on performance. We can also be conservative as you suggest, and do the check for all of the above to be on the safe side